airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/AcceptanceTests.java

```diff
@@ -407,18 +412,65 @@ public void testScheduledSync() throws Exception {
 
     createConnection(connectionName, sourceId, destinationId, catalog, connectionSchedule, syncMode);
 
     // When a new connection is created, Airbyte might sync it immediately (before the sync interval).
     // Then it will wait the sync interval.
     Thread.sleep(Duration.of(30, SECONDS).toMillis());
-    assertSourceAndTargetDbInSync(sourcePsql);
+    assertSourceAndTargetDbInSync(sourcePsql, false);
   }
 
   @Test
   @Order(10)
+  public void testIncrementalDedupSync() throws Exception {
+    final String connectionName = "test-connection";
+    final UUID sourceId = createPostgresSource().getSourceId();
+    final UUID destinationId = createDestination().getDestinationId();
+    final AirbyteCatalog catalog = discoverSourceSchema(sourceId);
+    final SyncMode syncMode = SyncMode.INCREMENTAL;
+    final DestinationSyncMode destinationSyncMode = DestinationSyncMode.APPEND_DEDUP;
+    catalog.getStreams().forEach(s -> s.getConfig()
+        .syncMode(syncMode)
+        .cursorField(List.of(COLUMN_ID))
+        .destinationSyncMode(destinationSyncMode)
+        .primaryKey(List.of(List.of(COLUMN_NAME))));
+    final UUID connectionId = createConnection(connectionName, sourceId, destinationId, catalog, null, syncMode).getConnectionId();
+
+    // reset back to no data.
+    final JobInfoRead jobInfoRead = apiClient.getConnectionApi().resetConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), jobInfoRead.getJob());
+    assertDestinationContains(Collections.emptyList(), STREAM_NAME);
+
+    // sync from start
+    final JobInfoRead connectionSyncRead1 = apiClient.getConnectionApi()
+        .syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead1.getJob());
+
+    assertSourceAndTargetDbInSync(sourcePsql, true);
+
+    // add new records and run again.
+    final Database source = getDatabase(sourcePsql);
+    final List<JsonNode> expectedRecords = retrieveSourceRecords(source, STREAM_NAME);
+    expectedRecords.add(Jsons.jsonNode(ImmutableMap.builder().put(COLUMN_ID, 6).put(COLUMN_NAME, "sherif").build()));
+    expectedRecords.add(Jsons.jsonNode(ImmutableMap.builder().put(COLUMN_ID, 7).put(COLUMN_NAME, "chris").build()));
+    source.query(ctx -> ctx.execute("UPDATE id_and_name SET id=6 WHERE name='sherif'"));
+    source.query(ctx -> ctx.execute("INSERT INTO id_and_name(id, name) VALUES(7, 'chris')"));
+    final List<JsonNode> sourceRecords = retrieveSourceRecords(source, STREAM_NAME);
+    source.close();
+
+    final JobInfoRead connectionSyncRead2 = apiClient.getConnectionApi()
+        .syncConnection(new ConnectionIdRequestBody().connectionId(connectionId));
+    waitForSuccessfulJob(apiClient.getJobsApi(), connectionSyncRead2.getJob());
+
+    assertDestinationContains(expectedRecords, STREAM_NAME);
+    assertFinalDestinationContains(sourceRecords);
```

Comment:
can you distinguish these two by calling one raw and one normalized? I think that's what's going on here, right? the "final" table is the normalized one and the first method invocation is for the raw table? that's not super clear right now. I think you should also change the variable names to help with that. I think the `expectedRecords` are the `expectedRecordsRaw` and `sourceRecords` are `expectedRecordsNormalized` or something like that.